Setup ipa-server-container

amore17 opened the pull request.

Tiboris commented on .github/workflows/gating.yaml:
```diff
@@ -46,3 +46,67 @@ jobs:
+              echo 'RUN pip install dbus-python python-ldap'
+              echo 'RUN python3 src/ipa-tuura/manage.py test'
+          } > podmanfile
+          podman build --tag gatingtest -f ./podmanfile
```
In general it is a Dockerfile or to be consistent Containerfile. It is not crucial just a cosmeting comment

Tiboris commented on .github/workflows/gating.yaml:
```diff
@@ -46,3 +46,67 @@ jobs:
+
+      - name: Pull bridge
+        run: |
+          sudo podman pull quay.io/idmops/bridge
```
here you can try to use https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry 
but I have not tried it yet. and push the image with the tag which is the PR number.

Tiboris commented on .github/workflows/gating.yaml:
```diff
@@ -46,3 +46,67 @@ jobs:
+          sudo podman exec ipa /usr/bin/bash -c 'ipactl restart'
+          sudo sleep 30
+          sudo podman exec ipa /usr/bin/bash -c 'echo Secret123 | kinit admin && echo vagrant'
+          sudo podman exec ipa /usr/bin/bash -c 'klist'
```
An example of usage of PR number in actions the `${{github.event.number}}` variable (NOT TESTED with latest changes): 
```yaml
build:
    runs-on: ubuntu-latest
    env:
      IMAGE_TAG: ${{ github.event.number || github.ref_name  }}
    name: build the docker image
    steps:
      - name: Checkout the repository
        uses: actions/checkout@v3

      - name: Set up QEMU for docker
        uses: docker/setup-qemu-action@v2

      - name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v2

      - name: Print IMAGE_TAG to be used
        run: echo "${{ env.IMAGE_TAG }}"

      - name: Print SECRETS as **** if they exist
        run: echo "username; ${{ secrets.REGISTRY_USERNAME }}"

      - name: Login to quay.io
        uses: docker/login-action@v2
        with:
          registry: quay.io
          username: ${{ secrets.REGISTRY_USERNAME }}
          password: ${{ secrets.REGISTRY_TOKEN }}

      - name: Build and push the image with a given tag
        uses: docker/build-push-action@v3
        with:
          context: ./src
          file: ./src/Containerfile 
          push: true
          tags: "quay.io/idmops/${{ github.event.repository.name }}:${{github.event.number}}"

  test:
    name: test the built image
    runs-on: ubuntu-latest
    container:
      image: "quay.io/idmops/${{ github.event.repository.name }}:${{ github.event.number || github.ref_name }}"  # here we can probably use GH registry and not quay as pull will not work or push for PRs
    needs: [pre-commit, test-on-ubuntu, build]

    steps:  # inside container
      - name: Display Container OS version
        run: cat /etc/redhat-release || uname -a

      - name: Run basic check for file presence
        run: file /ipa-tuura/manage.py || echo "/ipa-tuura/manage.py not found"

```

Tiboris reviewed (commented):
Some additional information from my experience @amore17 please use whatever will be useful for you.

f-trivino committed:
[wsgi] Add project path to the sys.path

When the WSGI server loads, Django needs to import the settings module by using
the DJANGO_SETTINGS_MODULE environment variable to locate the appropriate settings
module. It must contain the dotted path to the settings module.

This commit is adding the path to the settings module.

Related: https://github.com/freeipa/ipa-tuura/issues/29

Signed-off-by: Francisco Trivino <ftrivino@redhat.com>

f-trivino committed:
Add sample virtualhost .conf file for Apache2 on Fedora

This needs to be used when hosting the project in production with
Apache, mod_ssl, and mod_wsgi (enabling SSL).

Self-certificate can be generated with openssl.

Related: https://github.com/freeipa/ipa-tuura/issues/29
Signed-off-by: Francisco Trivino <ftrivino@redhat.com>

Tiboris commented:
you can try to rebase on commits from https://github.com/freeipa/ipa-tuura/pull/39 and we can continue building this

amore17 committed:
Test FREEIPA-9835

Signed-off-by: Anuja More <amore@redhat.com>

Tiboris commented on .github/workflows/gating.yaml:
```diff
@@ -122,3 +122,31 @@ jobs:
+      - name: Checkout sssd repository
+        uses: actions/checkout@v3
+        with:
+          path: sssd
```
I am afraid you miss a repository here:
```yaml
  with:
    repository: my-org/my-tools
    path: my-tools
```
in case of sssd repo I think it can be:
```yaml
  with:
    repository: SSSD/sssd-ci-containers
    path: sssd-ci-containers
```

Tiboris commented:
Also you need a tmp commit to change https://github.com/amore17/ipa-tuura/pull/1/commits/4c2ecd1c1229cc33d8c11a1fe4f09a69cd26f1cc#diff-4c81f8f5321c8b7181ad521b6ea16fe9feabbf0f7d0a0af6445cc2324df80df5R23 from freeipa to amore17

Tiboris commented on .github/workflows/gating.yaml:
```diff
@@ -112,14 +112,13 @@ jobs:
-          podman  pull ${{ env.IMAGE_REGISTRY_QUAY }}:${{ env.IMAGE_TAG }}
-          podman  images
+          sudo podman pull ${{ env.IMAGE_REGISTRY_QUAY }}:${{ env.IMAGE_TAG }}
+          sudo podman images
```
sudo should not be needed

Tiboris commented on .github/workflows/gating.yaml:
```diff
@@ -112,14 +112,13 @@ jobs:
+          sudo podman images
 
       - name: Pull the image previously built from Quay.io
-        if: github.event_name != 'pull_request'
```
this line is needed

Tiboris commented:
> You can now rebase on master of ipa-tuura, please keep project name change in temp commit
 
please do not mind my message, ignore it, my browser did not update status of your PR

f-trivino committed:
Fix project settings to allow deployment in production

These changes allow the project to be deployed in production by setting
two environment variables:

DJANGO_SECRET_KEY and DJANGO_DEBUG.

Defaults are intended for devel deployment. For production, the envvars
must be set accordingly:

https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Deployment

Fixes: https://github.com/freeipa/ipa-tuura/issues/33
Signed-off-by: Francisco Trivino <ftrivino@redhat.com>

Tiboris committed:
chore(GitHubActions): do not run registry interaction when PR is from fork

Signed-off-by: Tibor Dudlák <tdudlak@redhat.com>

Tiboris committed:
chore(GitHubActions): fix the condition to not skip push where it is needed

Signed-off-by: Tibor Dudlák <tdudlak@redhat.com>

Tiboris committed:
chore(GitHubActions): fix the condition so it does not use env var

Signed-off-by: Tibor Dudlák <tdudlak@redhat.com>

antoniotorresm committed:
Add containerization for the ipa-tuura+Apache service

Add Containerfile and podman-compose files that containerize ipa-tuura
using Apache as HTTPS server.

By running the podman-compose.yaml, the container is built and deployed
with the needed port mapping and environment variables.

Related: #29
Signed-off-by: Antonio Torres <antorres@redhat.com>
